src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -1,131 +1,311 @@
+    std::printf("    -o <path>: write the filtered signal into <path>. Notice you need to include filename\n");
+    std::printf("-h --help: print this help message.\n");
+    std::printf("Examples:\n");
+    std::printf("- Read 'fake_id.wav' and convolve with `IR.wav` using FFT convolver, with wet gain 0.1,\n");
+    std::printf("  FFT size 16384, write the filtered signal into 'out.wav':\n");
+    std::printf("  `MUSI6106Exec.exe -t freq -g 0.1 -b 16384 -i fake_id.wav -r IR.wav -o out.wav`\n");
+    std::printf("- Display this help:\n");
+    std::printf("  `MUSI6106Exec.exe -h`\n");
+    std::printf("  or\n");
+    std::printf("  `MUSI6106Exec.exe --help`\n");
+}
 
-    std::string             sInputFilePath,                 //!< file paths
-        sOutputFilePath;
+/*
+Simple command line argument parser.
+*/
+ConvolverArgs_t parseArg(int argc, char** argv) {
+    ConvolverArgs_t result;
+    for (int i = 0; i < argc; ++i) {
+        if (argv[i][0] == '-') { /* this is an argument */
+            if (argv[i][1] == 't') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of comb filter type!"));
+                ++i;
+                if (!strcmp(argv[i], "time")) {
+                    result.convolver = CFastConv::ConvCompMode_t::kTimeDomain;
+                }
+                else if (!strcmp(argv[i], "freq")) {
+                    result.convolver = CFastConv::ConvCompMode_t::kFreqDomain;
+                }
+                else {
+                    throw(std::invalid_argument("unknown argument!"));
+                }
+            }
+            else if (argv[i][1] == 'g') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of wet gain!"));
+                result.wetGain = std::stof(argv[++i]);
+            }
+            else if (argv[i][1] == 'b') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of block size!"));
+                result.blockSize = std::stoi(argv[++i]);
+            }
+            else if (argv[i][1] == 'i') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of input file path!"));
+                result.inputPath = std::string(argv[++i]);
+            }
+            else if (argv[i][1] == 'r') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of IR file path!"));
+                result.IRPath = std::string(argv[++i]);
+            }
+            else if (argv[i][1] == 'o') {
+                if (i == argc - 1) throw(std::invalid_argument("Incomplete argument of output file path!"));
+                result.outputPath = std::string(argv[++i]);
+            }
+            else if (argv[i][1] == 'h' || !strcmp(argv[i], "--help")) {
+                printUsage();
+                exit(0);
+            }
+            else {
+                throw(std::invalid_argument("unknown argument!"));
```

Comment:
I think it would be more helpful as long as there's an exception thrown. Maybe a try-block outside printing `e.what()` and usage would be easier to implement meanwhile more informative.

Reply:
Implemented in 124e0ca42befb37ca1333b93919d3b4ec95147a3.